cli/internal/codegen/ts.go

```diff
@@ -310,7 +310,7 @@ func (ts *ts) writeBaseClient() {
         let response = await fetch(this.baseURL + path, {
             method: method,
             headers: this.headers,
-            body: JSON.stringify(req || {})
+            body: JSON.stringify(req)
```

Comment:
This works but the behavior that `JSON.stringify(undefined) === undefined` is not very intuitive to me. If I saw this code somewhere "in the wild" I would have guessed that there's a bug lurking somewhere.

I think a semantically equivalent but clearer change would be to do `req !== undefined ? JSON.stringify(req) : undefined` or similar. That way the fact that we're dealing with the case when `req` is undefined is made explicit.

(Same goes for all the other places below)

Reply:
Good point! I decided to go with the shorter but equal expression `req && JSON.stringify(req)`.

Reply:
They're not equivalent; there are [lots of values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) that are falsy in JS. We only want the `undefined` value to indicate that the request should be sent without a body, not any falsy value.

Let's go with the explicit and "clearly correct" `req !== undefined ? JSON.stringify(req) : undefined` over trying to find a shorter, equivalent expression that will puzzle future readers as to what the intent is.

Reply:
Sure, forgot about those that were still JSON-stringifyable and falsy! New commit pushed!